src/components/views/rooms/RoomSettings.js

```diff
@@ -141,11 +268,96 @@ module.exports = React.createClass({
         });
     },
 
+    onAliasChanged: function(domain, index, alias) {
+        if (alias === "") return; // hit the delete button to delete please
+        var oldAlias;
+        if (this.isAliasValid(alias)) {
+            oldAlias = this.state.aliases[domain][index];
+            this.state.aliases[domain][index] = alias;
+            this.setState({ aliases_changed : true });
+        }
+        else {
+            var ErrorDialog = sdk.getComponent("dialogs.ErrorDialog");            
+            Modal.createDialog(ErrorDialog, {
+                title: "Invalid alias format", 
+                description: "'" + alias + "' is not a valid format for an alias",
+            });
+        }        
+    },
+
+    onAliasDeleted: function(domain, index) {
+        // XXX: can we edit state directly and then set, or should we copy it first?
+        var alias = this.state.aliases[domain].splice(index, 1);
```

Comment:
You can do this but don't get into the habit of it. React will whine loudly at direct `this.state` manipulations, but AFAIK it won't on this because it isn't savvy enough to know that `.splice` will modify the state object. There's nothing wrong with what you're doing currently.